appveyor_before_build.ps1

```diff
@@ -0,0 +1,21 @@
+. ./appveyor_helpers.ps1
+
+#
+# Copies the C# plugin so that the SonarQube MSBuild Scanner packaging projects will patch it. The packaging project is invoked by Appveyor as part of the regular build.
+#
+function CopyCsharpPluginForPatching
+{
+    Add-AppveyorMessage -Message "Copying the C# plugin for patching"
+        
+    $csPluginCleanJar = FindSingleFile "$mavenLocalRepository\org\sonarsource\dotnet\sonar-csharp-plugin" "*SNAPSHOT.jar"
+   
+    $destinationFile = [System.IO.Path]::Combine($env:APPVEYOR_BUILD_FOLDER, "PackagingProjects\CSharpPluginPayload", "csharp_plugin.jar");
+    [System.IO.File]::Copy($csPluginCleanJar, $destinationFile)
+}
+
+
+Add-AppveyorMessage -Message "Building the latest working C# plugin"
+DownloadAndBuildFromGitHub "SonarSource/sonar-csharp" "master"
```

Comment:
This relies on a GitHub feature that allows download of a repo snapshot as a zip.